xarray/core/dataset.py

```diff
@@ -6116,6 +6116,45 @@ def to_array(
 
         return DataArray._construct_direct(variable, coords, name, indexes)
 
+    def to_datatree(self, node_name: str | None = None):
+        """
+        Convert this dataset into a datatree.DataTree.
+
+        .. warning:: The DataTree structure is considered experimental,
+            and the API is less solidified than for other xarray features.
```

Comment:
not sure if I just don't know enough about rst, but I wonder if it would be better to move the whole text into the block?
```suggestion
        .. warning::
            The DataTree structure is considered experimental, and the API
            is less solidified than for other xarray features.
```